pkg/datarepair/checker/checker.go

```diff
@@ -127,8 +141,19 @@ func (checker *Checker) IdentifyInjuredSegments(ctx context.Context) (err error)
 
 					// reset durability stats for next iteration
 					checker.monStats = durabilityStats{}
+
+					if checker.pieceTracker != nil {
+						err := checker.gcService.Send(ctx, checker.pieceTracker, func() {
+							checker.gcWaitGroup.Done()
+						})
+						if err != nil {
+							checker.logger.Sugar().Errorf("error sending from garbage service: %v", err)
+						}
+					}
+					checker.pieceTracker = checker.gcService.NewPieceTracker()
 				}
 			}()
+			checker.gcWaitGroup.Add(1)
```

Comment:
the call to `gcWaitGroup.Wait` will deadlock if there is no piecetracker or pieces